common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -1155,4 +1166,30 @@ private void verifyTokenParametersAreSupported(@NonNull final TokenCommandParame
                             + " Current required version is set to: " + parameters.getRequiredBrokerProtocolVersion());
         }
     }
+
+    /**
+     * Check if have received broker version not supported error.
+     * @param resultBundle Result bundle from a broker operation.
+     * @param requiredBrokerProtocolVersion Required broker protocol version sent in request.
+     * @throws UnsupportedBrokerException if result contains broker not supported error.
+     */
+    private void verifyBrokerVersionIsSupported(@Nullable final Bundle resultBundle, @Nullable final String requiredBrokerProtocolVersion) throws UnsupportedBrokerException {
+        final String methodTag = TAG + ":verifyBrokerVersionIsSupported";
+        if (resultBundle == null) {
+            Logger.info(methodTag, "result bundle is null");
+            return;
+        }
+
+        // check if result bundle contains unsupported broker version exception
+        try {
+            final BrokerResult brokerResult = mResultAdapter.brokerResultFromBundle(resultBundle);
+            if (!brokerResult.isSuccess()
+                    && ErrorStrings.UNSUPPORTED_BROKER_VERSION_ERROR_CODE.equals(brokerResult.getErrorCode())) {
+                mHelloCache.saveHandshakeError(requiredBrokerProtocolVersion, CLIENT_MAX_PROTOCOL_VERSION);
+                throw new UnsupportedBrokerException(mActiveBrokerPackageName);
+            }
+        } catch (final ClientException e) {
+            Logger.error(methodTag, "Unable to read broker result from result bundle", e);
```

Comment:
Do we need to throw the exception back?

Reply:
Like I said in other comment, each operation has different format on result.

Only handling version check here